SwiftCompilerSources/Sources/SIL/Test.swift

```diff
@@ -0,0 +1,136 @@
+//===----------- Test.swift - In-IR tests from Swift source ---------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Basic
+import SILBridging
+
+@_semantics("arc.immortal")
+public class FunctionTest {
+  public var name: String
+  init(name: String, impl: @convention(thin) (Function, TestArguments, FunctionTest) -> ()) {
+    self.name = name
+    self.name._withStringRef { nameStr in
+      registerFunctionTest(
+        nameStr, 
+        unsafeBitCast(impl, to: UnsafeMutableRawPointer.self))
+      { bridgedFunction, bridgedTestArguments, bridgedFunctionTest, ctx in
+        typealias Impl = @convention(thin) (Function, TestArguments, FunctionTest) -> ()
+        let impl = unsafeBitCast(ctx, to: Impl.self)
+        impl(bridgedFunction.function, bridgedTestArguments.native, bridgedFunctionTest.native)
+      }
+    }
+  }
+  public var bridged: BridgedFunctionTest { BridgedFunctionTest(obj: SwiftObject(self)) }
+}
+
+extension BridgedFunctionTest {
+  public var native: FunctionTest { obj.getAs(FunctionTest.self) }
+}
+
+@_semantics("arc.immortal")
+public class TestArguments {
```

Comment:
again, does this need to be a bridged class?

Reply:
Nope, fixed.